controllers/asosecret_controller.go

```diff
@@ -279,22 +279,25 @@ func (asos *ASOSecretReconciler) createSecretFromClusterIdentity(ctx context.Con
 	newASOSecret.Data["AZURE_CLIENT_ID"] = []byte(identity.Spec.ClientID)
 
 	// Fetch identity secret, if it exists
-	key = types.NamespacedName{
-		Namespace: identity.Spec.ClientSecret.Namespace,
-		Name:      identity.Spec.ClientSecret.Name,
-	}
-	identitySecret := &corev1.Secret{}
-	err := asos.Get(ctx, key, identitySecret)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return nil, errors.Wrap(err, "failed to fetch AzureClusterIdentity secret")
-	}
+	if identity.Spec.ClientSecret != (corev1.SecretReference{}) {
```

Comment:
non-blocking, but I slightly prefer this for the cherry-pick, reads for easily as "not empty" to me, might be a personal preference though:

```suggestion
		if identity.Spec.ClientSecret.Name != "" {
```

Reply:
I'll do that.